node_modules/oae-tenants/lib/api.js

```diff
@@ -132,15 +132,23 @@ var init = module.exports.init = function(_serverConfig, callback) {
  * Get a list of all available tenants from cache. The global admin tenant will be excluded from the resulting tenant list
  *
  * @param  {Boolean}        [excludeDisabled]   Whether or not disabled tenants should be included. By default, all tenants will be returned
+ * @param  {Number}         [start]             The index to start paging at, defaults to 0
+ * @param  {Number}         [limit]             The maximum number of tenants to return, if not specified all tenants will be returned
  * @return {Object}                             An object keyed by tenant alias holding all the tenants
  */
-var getTenants = module.exports.getTenants = function(excludeDisabled) {
+var getTenants = module.exports.getTenants = function(excludeDisabled, start, limit) {
     var filteredTenants = {};
-    _.each(tenants, function(tenant, tenantAlias) {
-        // Exclude all disabled tenants when `exludeDisabled` has been provided
-        if (!tenant.isGlobalAdminServer && (!excludeDisabled || tenant.active)) {
-            filteredTenants[tenantAlias] = _copyTenant(tenant);
-        }
+    start = start || 0;
+    // Filter out the admin tenant and disabled tenants if requested
+    var preFiltered = excludeDisabled? _.where(tenants, {active: true, isGlobalAdminServer: false}): _.where(tenants, {isGlobalAdminServer: false});
+    preFiltered = _.sortBy(preFiltered, 'alias');
```

Comment:
As we're going to have 20k objects in here, it might make sense to have these cached and pre-sorted somewhere?